superset-frontend/src/chart/chartAction.js

```diff
@@ -258,41 +331,31 @@ export function exploreJSON(
     const logStart = Logger.getTimestamp();
     const controller = new AbortController();
 
-    const useLegacyApi = shouldUseLegacyApi(formData);
-
-    let requestParams = {
+    const requestParams = {
       signal: controller.signal,
       timeout: timeout * 1000,
     };
+    if (dashboardId) requestParams.dashboard_id = dashboardId;
 
-    if (allowDomainSharding) {
-      requestParams = {
-        ...requestParams,
-        mode: 'cors',
-        credentials: 'include',
-      };
-    }
-
-    const queryPromiseRaw = useLegacyApi
-      ? legacyChartDataRequest(
-          formData,
-          force,
-          method,
-          dashboardId,
-          requestParams,
-        )
-      : v1ChartDataRequest(formData, force, requestParams);
```

Comment:
I set up `legacyChartDataRequest` and `v1ChartDataRequest` to have different function signatures on purpose. I think it's better to keep `v1ChartDataRequest` simpler rather than conforming to a legacy set of parameters.

Feel free to keep the change if you prefer. Just wanted to state my original intent there.

Reply:
Good point.. The original reason I harmonized the signatures is that the new request function ended up requiring additional parameters that were in line with the legacy request (`force` and `requestParams`), so I ended up throwing `method` in there to make the signatures equal. But you're completely right that there's really no reason why they need to be equal, and going forward I anticipate the next version will probably not be backwards compatible with this request. So I'll either drop `method` from the new request or add a comment to explain that they've been made equal for convenience purposes only, and that it isn't strictly required.

Reply:
I agree with your assessment @suddjian , no point in making the signatures identical. I dropped `method` from `v1ChartDataRequest` to make sure we don't unnecessarily hang on to legacy constructs in the codebase.